esti/gc_test.go

```diff
@@ -0,0 +1,217 @@
+package esti
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/spf13/viper"
+	"github.com/treeverse/lakefs/pkg/api"
+	"github.com/treeverse/lakefs/pkg/block"
+)
+
+const (
+	fullGCMode          = 0
+	markOnlyMode        = 1
+	sweepOnlyMode       = 2
+	committedGCRepoName = "gc"
+	gcFindingsFilename  = "gc-findings.json"
+)
+
+var (
+	currentEpochInSeconds = time.Now().Unix()
+	dayInSeconds          = int64(100000) // rounded up from 86400
+)
+
+type testCase struct {
+	id           string
+	policy       api.GarbageCollectionRules
+	branches     []branchProperty
+	fileDeleted  bool
+	description  string
+	directUpload bool
+	testMode     string
+	markId       string
+}
+
+type branchProperty struct {
+	name                string
+	deleteCommitDaysAgo int64
+}
+
+var testCases = []testCase{
+	{id: "1",
+		policy:       api.GarbageCollectionRules{Branches: []api.GarbageCollectionRule{}, DefaultRetentionDays: 1},
+		branches:     []branchProperty{{name: "a1", deleteCommitDaysAgo: 2}, {name: "b1", deleteCommitDaysAgo: 2}},
+		fileDeleted:  true,
+		description:  "The file is deleted according to the default retention policy",
+		directUpload: false,
+	},
+}
+
+func TestCommittedGC(t *testing.T) {
+	SkipTestIfAskedTo(t)
+	blockstoreType := viper.GetViper().GetString("blockstore_type")
+	//TODO lynn: change this for test also on Azure
+	if blockstoreType != block.BlockstoreTypeS3 {
+		t.Skip("Running on S3 only")
+	}
+
+	step := viper.GetViper().GetString("gc_step")
+	switch step {
+	case "pre":
+		t.Run("pre", testPreCommittedGC)
+	case "post":
+		t.Run("post", testPostCommittedGC)
+	default:
+		t.Skip("No known value (pre/port) for gc_step")
+	}
+}
+
+func testPreCommittedGC(t *testing.T) {
+	ctx := context.Background()
+	blockstoreType := viper.GetViper().GetString("blockstore_type")
+	fileRefPerTestCase := make(map[string]string)
+
+	for _, testCase := range testCases {
+		t.Logf("Test case %s", testCase.id)
```

Comment:
Can we do this in `t.Run`?
```suggestion
		t.Run(fmt.Sprintf("Test case %s", testCase.id), func (t *testing.T) {
```